rest_docs/rest-catalog-open-api.yaml

```diff
@@ -0,0 +1,849 @@
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/responses/IcebergErrorResponse'
+              examples:
+                NamespaceNotFound:
+                  $ref: '#/components/examples/NoSuchNamespaceError'
+        406:
+          $ref: '#/components/responses/UnsupportedOperationResponse'
+        422:
+          description: Unprocessable Entity - A property key was included in both `removals` and `updates`
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/responses/IcebergErrorResponse'
+              examples:
+                UnprocessableEntityDuplicateKey:
+                  $ref: '#/components/examples/UnprocessableEntityDuplicateKey'
+
+  /v1/namespaces/{namespace}/tables:
+    parameters:
+      - $ref: '#/components/parameters/namespace'
+    get:
+      tags:
+        - Catalog API
+      summary: List all table identifiers underneath a given namespace
+      description: Return all table identifiers under this namespace
+      operationId: listTables
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/responses/ListTablesResponse'
+              examples:
+                NonEmptyResponse:
+                  $ref: '#/components/examples/ListTablesNonEmptyExample'
+                EmptyNamespaceResponse:
+                  $ref: '#/components/examples/ListTablesEmptyExample'
+        401:
+          $ref: '#/components/responses/UnauthorizedResponse'
+        404:
+          description: Not Found - The namespace specified does not exist
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/responses/IcebergErrorResponse'
+              examples:
+                NamespaceNotFound:
+                  $ref: '#/components/examples/NoSuchNamespaceError'
+
+  /v1/namespaces/{namespace}/tables/{table}:
+    parameters:
+      - $ref: '#/components/parameters/namespace'
+      - $ref: '#/components/parameters/table'
+    delete:
+      tags:
+        - Catalog API
+      summary: Drop a table from the catalog
+      operationId: dropTable
+      description: Remove a table from the catalog
+      parameters:
+        - name: purgeRequested
+          in: query
+          required: false
+          description: Whether the user requested to purge the underlying table's data and metadata
+          schema:
+            type: boolean
+            default: false
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/responses/DropTableResponse'
+              example: { "dropped": true, "purged": false }
+        202:
+          description: Accepted - for use if purgeRequested is implemented as an asynchronous action.
```

Comment:
Just curious, any way to check if async action finished?

Reply:
Yeah. At present we've left it out for simplicity (as there's already been a lot of discussion around just this), so it will be include it in a follow up PR, but there can be a callback URL that a client can subscribe to:

https://swagger.io/docs/specification/callbacks/